### ISSUE TITLE
jsdom npm package breaks bundled code of inflight function

### ISSUE THREAD

**Issue by skorfmann** [winglang/wing#2847]
### I tried this:

install the https://github.com/jsdom/jsdom package

`npm install jsdom`

```
// bar.mjs
import { JSDOM } from 'jsdom';

export async function parseFoo() {
  return 'bar';
}
```

```
// main.w
class Bar {
  extern "./bar.mjs" inflight parseFoo(): str;
  init() { }
}


test "it compiles" {
  bar.parseFoo();
}
```

### This happened:

```
wing test main.w
fail ┌ main.wsim » root/env0/test:parse atom feed
     │ /var/folders/zl/z8nc41qs7010qsfngfc_z1zc0000gn/T/wing-bundles-7V8N5O/index.js:28238
     │           fn = createFunction(`with (arguments[0]) { return function onerror(event, source, lineno, colno, error) {
     │                                ^^^^
     │
     │ SyntaxError: Unexpected token 'with'
     │     at new Script (node:vm:100:7)
     │     at createScript (node:vm:259:10)
     │     at Object.runInContext (node:vm:290:10)
```

it works when removing the `jsdom` import.

### I expected this:

In the best case it would work without an error 

### Component

Compiler

### Wing Version

0.19.2

### Node.js Version

v18.16.0

### Platform(s)

MacOS

### Anything else?

Not sure if that's a supported case, since `jsdom` is quite a unique and heavy package. On the other hand, people will probably want to run stuff like this or puppeteer / playwright.

Things I've observed, not sure if related:

- `jsdom` is using [vm](https://github.com/jsdom/jsdom#interfacing-with-the-nodejs-vm-module-using-getinternalvmcontext) as well
- [with](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/with) is deprecated and forbidden in strict mode

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by ekeren** [comment]
keep

**Comment by staycoolcall911** [comment]
TL;DR: Tried it out now. Still failing, but with a different error message. I'm not sure it's useful to keep this issue so I'm closing it.

I had to modernize the code above at `main.w` to:
```ts
// main.w
class Bar {
  pub extern "./bar.mjs" static inflight parseFoo(): str;
  new() { }
}

test "it compiles" {
  Bar.parseFoo();
}
```

Then I got this error message when running `wing test main.w`:
```sh
▲ [WARNING] "./xhr-sync-worker.js" should be marked as external for use with "require.resolve" [require-resolve-not-external]

    node_modules/jsdom/lib/jsdom/living/xhr/XMLHttpRequest-impl.js:31:57:
      31 │ const syncWorkerFile = require.resolve ? require.resolve("./xhr-sync-worker.js") : null;
         ╵                                                          ~~~~~~~~~~~~~~~~~~~~~~

fail ┌ main.wsim » root/env0/test:it compiles
     │ Error: Cannot find module './xhr-sync-worker.js'
     │ Require stack:
     │ - /Users/urib/Github/tries/stam/target/test/main.wsim/.wing/handler_c8224953.sandbox.js.bundle/index.js
     │    --> target/test/main.wsim/.wing/handler_c8224953.sandbox.js.bundle/index.js:13:50
     │    |   return fn && (res = (0, fn[__getOwnPropNames(fn)[0]])(fn = 0)), res;
     │    | };
     │    | var __commonJS = (cb, mod) => function __require() {
     │ 13 |   return mod || (0, cb[__getOwnPropNames(cb)[0]])((mod = { exports: {} }).exports, mod), mod.exports;
     │    |                                                  ^
     │ at __require /Users/urib/Github/tries/stam/target/test/main.wsim/.wing/handler_c8224953.sandbox.js.bundle/index.js:13:50
     │ at __require /Users/urib/Github/tries/stam/target/test/main.wsim/.wing/handler_c8224953.sandbox.js.bundle/index.js:13:50
     │ at __require /Users/urib/Github/tries/stam/target/test/main.wsim/.wing/handler_c8224953.sandbox.js.bundle/index.js:13:50
     └ at __require /Users/urib/Github/tries/stam/target/test/main.wsim/.wing/handler_c8224953.sandbox.js.bundle/index.js:13:50
 
 
Tests 1 failed (1)
Test Files 1 failed (1)
Duration 0m0.96s
```

Seems to be an issue with `jsdom` - I'm not sure if it makes sense to keep this issue open. I'll go ahead and close it. If anyone feels we should keep this issue, please feel free to reopen.
